ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util/LeakyBucket.hs

```diff
@@ -123,43 +132,43 @@ evalAgainstBucket config action = fst <$> runAgainstBucket config action
 -- the action terminates. Exposed for testing purposes.
 runAgainstBucket ::
   forall m a.
-  (MonadDelay m, MonadAsync m, MonadFork m, MonadMask m) =>
+  (MonadDelay m, MonadAsync m, MonadFork m, MonadMask m, NoThunks (m ())) =>
   Config m ->
   (Handlers m -> m a) ->
   m (State (Config m), a)
 runAgainstBucket config action = do
   bucket <- init config
   tid <- myThreadId
-  thread <- uncheckedNewTVarM Nothing
-  startThread thread bucket tid
+  killThreadVar <- newTVarIO Nothing
+  startThread killThreadVar bucket tid
   finally
```

Comment:
Not really related to this PR, and very minor, but this isn't async exception safe if the exception is thrown just after `startThread killThreadVar bucket tid`, but before the `finally` block.

Reply:
Indeed, and it used to be that I had to for variable scoping reasons but that's not the case anymore, so a4387acf7e1e029f97afdbc826d2db9ee7b5d9a3.